<?xml version="1.0" encoding="UTF-8"?>
<interface>
  <requires lib="gtk+" version="3.20"/>
  <template class="CcKeyboardShortcutEditor" parent="GtkDialog">
    <property name="can_focus">False</property>
    <property name="resizable">False</property>
    <property name="modal">True</property>
    <property name="width_request">400</property>
    <property name="height_request">300</property>
    <property name="window_position">center</property>
    <property name="type_hint">dialog</property>
    <signal name="close" handler="cancel_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="no" />
    <signal name="delete-event" handler="gtk_widget_hide_on_delete" object="CcKeyboardShortcutEditor" swapped="yes"/>
    <child internal-child="vbox">
      <object class="GtkBox">
        <property name="can_focus">False</property>
        <property name="orientation">vertical</property>
        <property name="spacing">18</property>
        <property name="border_width">12</property>
        <child>
          <object class="GtkLabel" id="top_info_label">
            <property name="visible">True</property>
            <property name="can_focus">False</property>
            <property name="wrap">True</property>
            <property name="wrap_mode">word-char</property>
            <property name="width_chars">15</property>
            <property name="max_width_chars">20</property>
          </object>
          <packing>
            <property name="position">0</property>
          </packing>
        </child>
        <child>
          <object class="GtkStack" id="stack">
            <property name="visible">True</property>
            <property name="can_focus">False</property>
            <property name="hexpand">True</property>
            <property name="vexpand">True</property>
            <child>
              <object class="GtkBox" id="edit_box">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="orientation">vertical</property>
                <property name="spacing">18</property>
                <property name="expand">True</property>
                <property name="halign">center</property>
                <property name="valign">center</property>
                <child>
                  <object class="GtkImage">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="resource">/org/bamboo/keyboard-editor/enter-keyboard-shortcut.svg</property>
                  </object>
                </child>
                <child>
                  <object class="GtkLabel">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="wrap">True</property>
                    <property name="label" translatable="yes">Press Esc to cancel.</property>
                    <style>
                      <class name="dim-label" />
                    </style>
                  </object>
                </child>
              </object>
            </child>
            <child>
              <object class="GtkBox" id="standard_box">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="hexpand">True</property>
                <property name="orientation">vertical</property>
                <property name="spacing">18</property>
                <child>
                  <object class="GtkBox">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="spacing">18</property>
                    <child type="center">
                      <object class="GtkShortcutLabel" id="shortcut_accel_label">
                        <property name="visible">True</property>
                        <property name="can_focus">False</property>
                        <property name="halign">center</property>
                        <property name="disabled-text" translatable="yes">Disabled</property>
                      </object>
                    </child>
                    <child>
                      <object class="GtkButton" id="clear_button">
                        <property name="visible">True</property>
                        <property name="can_focus">True</property>
                        <property name="relief">none</property>
                        <property name="halign">end</property>
                        <child>
                          <object class="GtkImage">
                            <property name="visible">True</property>
                            <property name="can_focus">False</property>
                            <property name="icon-name">edit-clear-symbolic</property>
                          </object>
                        </child>
                        <signal name="clicked" handler="clear_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
                      </object>
                      <packing>
                        <property name="pack_type">end</property>
                      </packing>
                    </child>
                  </object>
                </child>
                <child>
                  <object class="GtkLabel" id="shortcut_conflict_label">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="hexpand">True</property>
                    <property name="wrap">True</property>
                    <property name="wrap_mode">word-char</property>
                    <property name="width_chars">15</property>
                    <property name="max_width_chars">20</property>
                    <property name="xalign">0</property>
                  </object>
                </child>
              </object>
            </child>
          </object>
          <packing>
            <property name="expand">False</property>
            <property name="fill">True</property>
            <property name="position">1</property>
          </packing>
        </child>
      </object>
    </child>
    <child type="titlebar">
      <object class="GtkHeaderBar" id="headerbar">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <property name="show_close_button">True</property>
        <child>
          <object class="GtkButton" id="cancel_button">
            <property name="label" translatable="yes">Cancel</property>
            <property name="visible">True</property>
            <property name="can_focus">True</property>
            <property name="receives_default">True</property>
            <signal name="clicked" handler="cancel_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="no" />
          </object>
        </child>
        <child>
          <object class="GtkButton" id="revert_button">
            <property name="visible">True</property>
            <property name="can_focus">False</property>
            <property name="label" translatable="yes">Revert</property>
            <property name="valign">end</property>
            <signal name="clicked" handler="revert_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
          </object>
        </child>
        <child>
          <object class="GtkButton" id="set_button">
            <property name="label" translatable="yes">Set</property>
            <property name="visible">True</property>
            <property name="can_focus">True</property>
            <property name="can_default">True</property>
            <property name="receives_default">True</property>
            <signal name="clicked" handler="set_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
            <style>
              <class name="suggested-action" />
            </style>
          </object>
          <packing>
            <property name="pack_type">end</property>
            <property name="position">3</property>
          </packing>
        </child>
      </object>
    </child>
    <action-widgets>
      <action-widget response="cancel">cancel_button</action-widget>
      <action-widget response="apply">set_button</action-widget>
    </action-widgets>
  </template>
  <object class="GtkSizeGroup">
    <widgets>
      <widget name="cancel_button"/>
      <widget name="revert_button"/>
      <widget name="clear_button"/>
    </widgets>
  </object>
</interface>
